flink-core/src/main/java/org/apache/flink/api/common/io/BinaryInputFormat.java

```diff
@@ -213,9 +213,10 @@ protected SequentialStatistics createStatistics(List<FileStatus> files, FileBase
 
 			FSDataInputStream fdis = file.getPath().getFileSystem().open(file.getPath(), blockInfo.getInfoSize());
 			fdis.seek(file.getLen() - blockInfo.getInfoSize());
-			
+
 			blockInfo.read(new DataInputViewStreamWrapper(fdis));
 			totalCount += blockInfo.getAccumulatedRecordCount();
+			fdis.close();
```

Comment:
Agreed, we should try and consistently use a "try-with-resource" clause for streams.